google-cloud-spanner/lib/google/cloud/spanner/convert.rb

```diff
@@ -164,6 +103,64 @@ def raw_to_value obj
             end
           end
 
+          def field_for_raw obj
+            if NilClass === obj
```

Comment:
I think using Ruby's `case` is better here. `===` is not widely used in ruby community as it's asymmetric for ex.:
`Symbol === :symbol # => true`
`:symbol === Symbol # => false` 😱

I recommend doing something like:
```
case obj
when NilClass
   raise ArgumentError, "Cannot determine type for nil values."
when Date
  ................................
```

Reply:
This is generally true, but for IO-ish objects we need to detect if the object has `read` and `rewind` methods (`elsif obj.respond_to?(:read) && obj.respond_to?(:rewind)`), and we can't do that with a `case` statement.